### ISSUE TITLE
Chest Cavity Newer Version Compatibility Issue [Non-Major]

### ISSUE THREAD

**Issue by Orion-Wyatt** [eehunter/RiseOfTheToasters#1]
Apon Wanting to add the Chest Cavity mod after using only the Toaster mod, I tried adding it and ended being stuck with the game crashing on start-up. _(This is with using the latest curseforge release of this mod in 1.16 Minecraft)_
It only happens with versions of the Chest Cavity mod beyond version 2.10.2 (up until version 2.13.0 where the **game** version stops compatibility, _as expected_), so as tagged in the title, it's not a big concern.

If you do wanna update the 1.16 version of this mod, I've attached the crash report when attempting to see if the game will launch with one version above the last compatible version of the Chest Cavity mod (In an attempt to help, though I have no clue if it will).
[crash-2021-10-27_23.37.13-client.txt](https://github.com/eehunter/RiseOfTheToasters/files/7429853/crash-2021-10-27_23.37.13-client.txt)

If you don't plan on adding compatibility to the newer versions of Chest Cavity Mod (within 1.16 Minecraft), I'm sure others who try to use these two mods together wouldn't mind a note in the description mentioning this.

Thanks for taking your time to read this.
=^w^=

**Comment by eehunter** [comment]
Thanks for telling me. The 1.16.5 version of the mod (RotT 1.1.x) isn't really maintained anymore. The 1.17.1 version is going to be modular, and as of last night the first module (See my FurLib repo), the first module is nearly ready for an alpha release. I just need to tidy up the code a bit more and make sure everything works as expected. I'm not sure if I will maintain Chest Cavity integration in RotT 1.2.x, as I intend to use a custom cybernetics system based around Apoli (the library behind Origins). It shouldn't be incompatible, but currently I don't plan to have native integration. The new system will allow custom cyberware to be defined entirely using datapacks, and will have full access to Apoli powers. New PowerType instances can also be defined using Java or Kotlin, as usual. After I release FurLib (hopefully sometime in the text few days), I plan to start work on the Rise of the Toasters: Core and Rise of the Toasters: Cybernetics modules. The Core module will add basic crafting materials and likely API stuff used by other modules, and the Cybernetics module will add the limb system, as well as some other stuff. I will say that native CC integration isn't out of the question for the Cybernetics module, but it isn't a high priority, as there are other modules I want to work on (namely RotT: Exploration, which will have worldgen stuff). The Curseforge page for the legacy (1.16.5) versions will be used for RotT: Core, and new ones will be made for the other modules.

**Comment by emyfops** [comment]
> Thanks for telling me. The 1.16.5 version of the mod (RotT 1.1.x) isn't really maintained anymore. The 1.17.1 version is going to be modular, and as of last night the first module (See my FurLib repo), the first module is nearly ready for an alpha release. I just need to tidy up the code a bit more and make sure everything works as expected. I'm not sure if I will maintain Chest Cavity integration in RotT 1.2.x, as I intend to use a custom cybernetics system based around Apoli (the library behind Origins). It shouldn't be incompatible, but currently I don't plan to have native integration. The new system will allow custom cyberware to be defined entirely using datapacks, and will have full access to Apoli powers. New PowerType instances can also be defined using Java or Kotlin, as usual. After I release FurLib (hopefully sometime in the text few days), I plan to start work on the Rise of the Toasters: Core and Rise of the Toasters: Cybernetics modules. The Core module will add basic crafting materials and likely API stuff used by other modules, and the Cybernetics module will add the limb system, as well as some other stuff. I will say that native CC integration isn't out of the question for the Cybernetics module, but it isn't a high priority, as there are other modules I want to work on (namely RotT: Exploration, which will have worldgen stuff). The Curseforge page for the legacy (1.16.5) versions will be used for RotT: Core, and new ones will be made for the other modules.

The best should be to directly make the mod in 1.18 since it releases really soon and will be valid for 1 and a half year

**Comment by eehunter** [comment]
FurLib should (in theory) be largely cross compatible with 1.18 as soon as its dependencies (Apoli and Calio) are ported. I may need to make separate releases, but doing so should be mostly trivial (I hope). I'm pretty sure I wrote my previous comment before I was aware of how close 1.18 was to release. Unless 1.18 brings non-negligable changes to player rendering, Brigader (minecraft's command processing library), Apoli/Calio, or a few other minor things that I doubt will change, FurLib (in it's current state) should be able to run (though I might have to change something in the fabric.mod.json so it doesn't assume it isn't compatible when it actually is).

**Comment by eehunter** [comment]
After checking the Apoli repo, it seems to have already been ported to 1.18, so I can test compatibility immeditely when I get home. I should probably ask my robotics advisor if I can use the machine I'm on now for my modding projects, as it's tecnically the robotics club's laptop, and it would be good for me to practice programming when I have extra time at school. The language I use for modding (Kotlin; I've stopped using Java for modding since I've discovered that Kotlin is a lot more fun to use) is the same one I plan to use for programming the club's robots, so the skills would transfer well. If he says yes, that would give me a lot more time for coding.
